oblast/src/lib.rs

```diff
@@ -70,6 +148,20 @@ pub struct Scalar {
     value: blst_scalar,
 }
 
+impl Scalar {
+    /// Construct a `blst_scalar` instance of a `Fr` value from bytes in big-endian order.
+    /// Panics if the value is not in `Fr`.
+    pub fn from_fr_bytes(value: &[u8]) -> Self {
+        assert!(value.len() == 32);
+        let mut scalar = blst::blst_scalar::default();
+        unsafe {
+            blst::blst_scalar_from_bendian(&mut scalar, value.as_ptr());
+            assert!(blst::blst_scalar_fr_check(&scalar));
```

Comment:
Long-term we may want to make this a `debug_assert` for performance reasons, but I'm happy to leave it in for now